File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int curr = 0, maxP = 0, minP = prices[0];
        for(int i = 1; i < prices.size(); i++) {
            if(minP > prices[i]) {
                minP = prices[i];
            } else {
                maxP = max(maxP, prices[i] - minP);
            }
        }

        return maxP;
    }
};

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.js
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function (prices) {
    let currMin = Number.POSITIVE_INFINITY;
    let max = 0;

    for (let i = 0; i < prices.length; i++) {
        if (currMin > prices[i]) {
            currMin = prices[i];
        } else {
            max = Math.max(max, prices[i] - currMin);
        }
    }

    return max;
};